File: 0506.相对名次/0506-相对名次.java
class Solution {
    public String[] findRelativeRanks(int[] nums) {
        int n = nums.length;
        int[] nums1 = Arrays.copyOf(nums, n);
        String[] result = new String[n];
        
        Arrays.sort(nums1);
        Map<Integer, Integer> map = new HashMap<>();
       
        for (int i = 0; i < n; i++){
            map.put(nums[i], i);
    }
        // System.out.println(Arrays.toString(nums1));
        
        for (int i = n - 1; i >=0 ; i--){
            if (i == n - 1)
                result[map.get(nums1[i])] = "Gold Medal";
            else if (i == n - 2)
                result[map.get(nums1[i])] = "Silver Medal";
            else if (i == n - 3)
                result[map.get(nums1[i])] = "Bronze Medal";
            else
                result[map.get(nums1[i])] = String.valueOf(n - i);
        }
        return result;
    }
}

File: 0506.相对名次/0506-相对名次.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        p = [[score[i], i] for i in range(len(score))]

        p.sort(key = lambda x: -x[0])

        res = [0 for _ in score]
        for index, pair in enumerate(p):
            score, original_index = pair[0], pair[1]

            if index == 0:
                val = "Gold Medal"
            elif index == 1:
                val = "Silver Medal"
            elif index == 2:
                val = "Bronze Medal"
            else:
                val = str(index + 1)

            res[original_index] = val
        return res